cmd/simRecon/simRecon.go

```diff
@@ -52,32 +52,52 @@ func ReconstructSeq(newickInput string, fastaInput string, outputFilename string
-func SimRecon(rootFastaFile string, treeFile string, gp string, simOutFile string, leafOutFile string, reconOutFile string, accuracyOutFile string) {
+func SimRecon(rootFastaFile string, treeFile string, gp string, simOutFile string, leafOutFile string, reconOutFile string, accuracyOutFile string, option int, baseAccFile string) {
 	//TODO: this code will need to change drastically for sequences of varying lengths.
 	//The loop through the sequence is restricted by the length of a single fasta and the tot calculation will need to calculate the total number of bps
 	//ReconAccuracy calculates the total number of incorrectly reconstructed base pairs in a tree and returns a percentage of correct base calls
 	SimulateEvolve(rootFastaFile, treeFile, gp, simOutFile, leafOutFile)
 	ReconstructSeq(treeFile, leafOutFile, reconOutFile)
 
-	answer := reconstruct.ReconAccuracy(simOutFile, reconOutFile, leafOutFile, gp)
+	answer, byBaseAnswer := reconstruct.ReconAccuracy(simOutFile, reconOutFile, leafOutFile, gp, option)
 	out := fileio.EasyCreate(accuracyOutFile)
 	defer out.Close()
 
 	for name, accuracy := range answer {
 		fmt.Fprintf(out, "%s\t%f\n", name, accuracy)
 	}
+	if option == 1 && baseAccFile != "" {
+		baseAccOut := fileio.EasyCreate(baseAccFile)
+		defer baseAccOut.Close()
+		for species, baseAcc := range byBaseAnswer {
+			firstBase := species + " First Base"
+			secondBase := species + " Second Base"
+			thirdBase := species + " Third Base"
+			for b := range baseAcc {
+				if b == 0 {
+					fmt.Fprintf(baseAccOut, "%s\t%f\n", firstBase, baseAcc[b])
+				} else if b == 1 {
+					fmt.Fprintf(baseAccOut, "%s\t%f\n", secondBase, baseAcc[b])
+				} else {
+					fmt.Fprintf(baseAccOut, "%s\t%f\n", thirdBase, baseAcc[b])
+				}
+			}
+		}
+	}
 }
 
 func usage() {
 	fmt.Print(
-		"SimRecon takes in a root fasta file and a tree. It simulates evolution alone the tree, performs reconstruction based on the leaf nodes of the tree and then calculates accuracy at each node.\n" +
-			"simRecon <rootFile.fasta> <treeFile.txt> <genePred.gp> <simOutFile.fasta> <leafOutFile.fasta> <reconOutFile.fasta> <accuracyOutFile.txt> \n" +
+		"SimRecon takes in a root fasta file, a genePred, and a tree. It simulates evolution along the tree, performs reconstruction based on the leaf nodes of the tree and then calculates accuracy at each node.\n" +
+			"simRecon [-option] <rootFile.fasta> <treeFile.txt> <genePred.gp> <simOutFile.fasta> <leafOutFile.fasta> <reconOutFile.fasta> <accuracyOutFile.txt>\n" +
 			"options:\n")
 	flag.PrintDefaults()
 }
 
 func main() {
 	var expectedNumArgs = 7
+	var option = flag.Int("option", 0, "Specify to 1 to return an accuracy breakdown by base position in codons. Default value is 0, which returns normal output.")
+	var baseAccFile = flag.String("baseAccFile", "", "Specify a filename for the output of accuracy by position of base in codon (if option = 1).")
```

Comment:
Do you need both baseAccFile and option?  It seems like you can't have one without the other and a test of baseAccFile against the empty string could act like checking if option equals 0/1.

Reply:
If you want to keep option, you may also consider making option a flag.Bool since the only acceptable values are 0 and 1.

Reply:
So maybe baseAccFile is not a file, but simply a name we want a future file to be? And in SimRecon, at the moment of input it is still a name and not actually a file? But then we never output a file. So is baseAccFile ever really a file? I'm likely missing something so I figured I'd write out my thoughts and maybe you can help me understand what I'm failing to comprehend. Thanks!